src/backend/joanie/lms_handler/backends/openedx.py

```diff
@@ -93,3 +114,54 @@ def set_enrollment(self, username, resource_link, active=True):
 
         logger.error(response.content)
         return None
+
+    def get_progression(self, username, resource_link=None):
+        """
+            Get state of each course runs delivering a certificate
+            to which username is enrolled
+
+            Return:
+            [
+                {
+                    "course_key": "course-v1:edX+DemoX+Demo_Course",
+                    "status": 
+                          ENROLLMENT_STATE_IN_PROGRESS
+                        | ENROLLMENT_STATE_VALIDATED
+                        | ENROLLMENT_STATE_FAILED,
+                    "grade": "0.98",
+                }
+            ]
+        """
+        base_url = self.configuration["BASE_URL"]
+
+        # We ensure compatiblity with OpenEdX Dogwood by passing two query parameters
+        # to send the username :
+        # - `query` parameter is for OpenEdX Dogwood
+        # - `user` parameter is for OpenEdX upper than Dogwood
+        response = self.api_client.request(
+            "GET", f"{base_url}/certificates/search?user={username}&query={username}"
+        )
+
+        if response.ok:
+            results = [
+                {
+                    "course_key": result["course_key"],
+                    "grade": result["grade"],
+                    "status": self._get_enrollment_state(result["status"]),
+                }
+                for result
+                in json.loads(response.content) if response.content else []
+            ] 
+
+            if resource_link:
+                course_id = self.extract_course_id(resource_link)
+                results = [
+                    result
+                    for result in results
+                    if result.get("course_key", "") == course_id
+                ]
```

Comment:
You can also use the builtin `filter` function:
```suggestion
                results = filter(lambda r: r.get("course_key", "") == course_id, results)
```